### PR TITLE
Printing collision info for invalid IK solutions.

### PR DESCRIPTION
**Pull request by NFiedler**
Instead of just marking an invalid IK solution in red, a comment is added describing the (and only one) collision causing the problem. 
This was done with a lot of help from @v4hn .

### CODE REVIEW

File: core/src/stages/compute_ik.cpp [hunk 1]
```diff
@@ -353,14 +360,25 @@ void ComputeIK::compute() {
...
 		ik_solutions.emplace_back();
-		state->copyJointGroupPositions(jmg, ik_solutions.back());
+		auto& solution{ ik_solutions.back() };
+		state->copyJointGroupPositions(jmg, solution.joint_positions);
+		collision_detection::CollisionRequest req;
+		collision_detection::CollisionResult res;
+		req.contacts = true;
+		req.max_contacts = 1;
+		scene->checkCollision(req, res, *state);
+		solution.feasible = ignore_collisions || !res.collision;
```
Comment (rhaschke) [inline core/src/stages/compute_ik.cpp:375]: When `ignore_collisions` is `true`, the collision checking should be skipped, shouldn't it?

File: core/src/stages/compute_ik.cpp [hunk 2]
```diff
@@ -388,15 +406,17 @@ void ComputeIK::compute() {
 			solution.setComment(s.comment());
 			std::copy(frame_markers.begin(), frame_markers.end(), std::back_inserter(solution.markers()));
 
-			if (succeeded && i + 1 == ik_solutions.size())
+			if (ik_solutions[i].feasible)
```
Comment (rhaschke) [inline core/src/stages/compute_ik.cpp:409]: I guess this new condition just aims to be more obvious.
Do we still need the variable `succeeded`?

**Review by rhaschke — CHANGES_REQUESTED** [review]
Looks good in general. But there is one critical remark:

File: core/src/stages/compute_ik.cpp (hunk 1 above)
Comment (v4hn) [reply core/src/stages/compute_ik.cpp:375]: We discussed tha when writing the patch and I am in favor of *always* doing collisions checks here to provide the feedback even if the solution is still marked as feasible. I notice now that checking was indeed skipped before in this case (as obviously no feedback was generated from it), but I would still change it in favor of the additional feedback.

If you insist it's better to retain the current behavior here in favor of performance we can do it as well, but `ignore_collisions=true` was always more of a debugging flag for me.

File: core/src/stages/compute_ik.cpp (hunk 2 above)
Comment (v4hn) [reply core/src/stages/compute_ik.cpp:409]: It's [used below](https://github.com/ros-planning/moveit_task_constructor/pull/376/files#diff-eb4791e6fc52e29c2b9fb96a606928a1bf587989c5ebf29af186a50137ecc159R434).

File: core/src/stages/compute_ik.cpp (hunk 1 above)
Comment (rhaschke) [reply core/src/stages/compute_ik.cpp:375]: You are right. The default case is to perform collision checking anyway. I just wanted to point out the difference in behavior.

**Review by rhaschke — APPROVED** [review]

**Comment by rhaschke** [comment]
Please fix the formatting issue before merging.

**Pull request merged**